### PR TITLE
Skeletal security context to facilitate tmpfs mount

### PR DESCRIPTION
**Pull request by pmorie**
Second shot at #6238, with less hacks and more stacks (of changed volume plugins)

@thockin @pweil-

### CODE REVIEW

**Comment by pmorie** [comment]
@thockin, @pweil- and I discussed this factoring as something that could be just enough of a security context sketch to be better than the version with rootdir, what do you think?

**Comment by pmorie** [comment]
@markturansky, any thought on this?

**Comment by pmorie** [comment]
@thockin @markturansky @pweil- @erictune @smarterclayton @csrwng sketched out a bit more.  If people like it, I'll wrap this up.

**Comment by thockin** [comment]
I'm OK with this.

**Comment by pweil-** [comment]
Fine with me.  I'll add the VolumeOptions to the SecurityContext types that I'll be resubmitting today for another review.

**Comment by pmorie** [comment]
I want #6166 to go in before this.

**Comment by pmorie** [comment]
@thockin @pweil- For e2e tests for this, i'm going to write to use the mount tester added in #6166 and establish that you cannot write to a secret file, but can read from it, etc.

**Comment by thockin** [comment]
So hold tight until tests are in place?  SGTM

**Comment by pmorie** [comment]
@thockin yep, should be later today

**Comment by pmorie** [comment]
@thockin @pweil- Rebased; I need to get an selinux consult on a failure I'm seeing locally

File: pkg/volume/empty_dir/empty_dir.go [hunk 1]
```diff
@@ -195,7 +198,29 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
 	if isMnt && medium == mediumMemory {
 		return nil // current state is what we expect
 	}
-	return ed.mounter.Mount("tmpfs", dir, "tmpfs", 0, "")
+	// If the directory is a mountpoint with medium memory, there is no
+	// work to do since we are already in the desired state.
+	if isMnt && medium == mediumMemory {
```
Comment (thockin) [inline pkg/volume/empty_dir/empty_dir.go]: Why is this block duplicated from above?

**Comment by thockin** [comment]
just a nit then LGTM.  holla when your consult is done.

File: pkg/volume/empty_dir/empty_dir.go (hunk 1 above)
Comment (pmorie) [inline pkg/volume/empty_dir/empty_dir.go]: Inadvertent, I will fix.

**Comment by pmorie** [comment]
@thockin I got my local policy issues cleared up and will rebase / squash later.

**Comment by pmorie** [comment]
@thockin Just remembered that I'm going to make some changes to e2e, too.

**Comment by pmorie** [comment]
@thockin PTAL

File: pkg/volume/secret/secret.go [hunk 2]
```diff
@@ -126,7 +127,7 @@ func (sv *secretVolume) SetUpAt(dir string) error {
 	for name, data := range secret.Data {
 		hostFilePath := path.Join(dir, name)
 		glog.V(3).Infof("Writing secret data %v/%v/%v (%v bytes) to host file %v", sv.podRef.Namespace, sv.secretName, len(data), hostFilePath)
-		err := ioutil.WriteFile(hostFilePath, data, 0777)
+		err := ioutil.WriteFile(hostFilePath, data, 0555)
```
Comment (pmorie) [inline pkg/volume/secret/secret.go]: @thockin I noticed that we were setting the perms foolishly -- this seems like a better mode.

Comment (thockin) [inline pkg/volume/secret/secret.go]: Do you really want it executable by deafult?  Why not 0444?

**Comment by thockin** [comment]
Just the one question, then LGTM

File: pkg/volume/secret/secret.go (hunk 2 above)
Comment (pmorie) [inline pkg/volume/secret/secret.go]: I've made it 0444 per our discussion on IRC

**Comment by pmorie** [comment]
@thockin: starting a gce run of this now, I have only tested locally

**Comment by thockin** [comment]
Will hang tight

**Comment by pmorie** [comment]
@thockin I have a clean GCE run of the secrets and emptyDir e2e tests; doing a full run now.

**Comment by pmorie** [comment]
@thockin I think this should be good.  I was most worried about the replication controller tests, but I believe that I had hit https://github.com/GoogleCloudPlatform/kubernetes/issues/6424.  I'm doing runs in master and gce now to compare.  Using an abundance of caution because I recently caused a breakage in network e2e with entrypoint control.

**Comment by thockin** [comment]
needs a rebase

**Comment by pmorie** [comment]
@thockin yep, rebased and the build was broken, will do another run

**Comment by thockin** [comment]
Needs a rebase (still or again)

**Comment by pmorie** [comment]
@thockin working on it -- seeing breakage in the secrets e2e after rebase
that I'm tracking down.

**Comment by pmorie** [comment]
@thockin I'm not sure what's going on on my local system, but I want to do another gce run before this gets merged.

**Comment by pmorie** [comment]
Got clean gce/local runs on emptyDir and secrets.

**Pull request merged**